### PR TITLE
DINFRA-1042: awscli, miniconda, conda env file added, Spark1.6 removed

### PR DESCRIPTION
**Pull request by ilyasemenov84**
In the scope of this PR `strider-docker-spark` has been actualised image modified to for the new CDP deployment:
- Group `jenkins` has been added to the image user and folder permission modified
- Spark 1.6 installation has been removed
- Python sensible libraries updated
- Miniconda added
- Conda env file added
- awscli added

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -44,12 +41,16 @@ RUN cd /opt/ \
   && make install \
   && rm -rf /opt/Python-3.7.4.tgz
 
+# update pip
+RUN pip install --upgrade pip==18.1
```
Comment (guidok91) [inline Dockerfile]: Do we need to use this older version of pip?

Comment (guidok91) [reply Dockerfile]: Can this affect current CI pipelines? (e.g. this image is the one used for all of them right?

Comment (ilyasemenov84) [reply Dockerfile]: Otherwise Ansible installation fails.
No, there is no impact, python3 and python2 existing pipelines are running fine.

Comment (ilyasemenov84) [reply Dockerfile]: Each ETL installation has its dedicated venv, also.

Comment (gmsantos) [reply Dockerfile]: I'm bundling ansible with etl-manager https://github.com/hellofresh/hf-etl-tool/blob/master/setup.py#L20 (just need to `pip install hf-etl-manager` and it's ready to go)

It is working just fine with python 3.7 and pip 20.1 venv. Can we drop in from the base image maybe?

Comment (ilyasemenov84) [reply Dockerfile]: 👍 let me try

File: Dockerfile [hunk 2]
```diff
@@ -27,10 +28,6 @@ RUN  apt-get update \
   libsasl2-dev \
   libsqlite3-dev \
   libbz2-dev
-RUN cd /opt/ \
```
Comment (gmsantos) [inline Dockerfile:30]: Should we lock the tag for previous pipelines that might still rely on Spark 1.6, or all pipelines are on spark 2.4?

If we merge it without the previous version, we need to lock the image tag on Python 2 pipelines.

File: hellofresh-py3-test.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+name: hellofresh-py3-test
+channels:
+  - conda-forge
+  - defaults
+dependencies:
+  - python=3.7
+  - pip:
+    - --index-url https://artifactory.tools-k8s.hellofresh.io/artifactory/api/pypi/py/simple
+    - biautomation==1.1.0
```
Comment (gmsantos) [inline hellofresh-py3-test.yml]:
```suggestion
    - biautomation>=2.0.0,<3.0.0
```

Grid Dynamics migrated biautomation to work on Python 3 [here](https://github.com/hellofresh/bi-automation/pull/29) (PR still open, but pack manually published to Artifactory).

File: hellofresh-py3.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+name: hellofresh-py3
+channels:
+  - conda-forge
+  - defaults
+dependencies:
+  - python=3.7
+  - pip:
+    - --index-url https://artifactory.tools-k8s.hellofresh.io/artifactory/api/pypi/py/simple
+    - biautomation==1.1.0
```
Comment (gmsantos) [inline hellofresh-py3.yml]:
```suggestion
    - biautomation>=2.0.0,<3.0.0
```

Grid Dynamics migrated biautomation to work on Python 3 [here](https://github.com/hellofresh/bi-automation/pull/29) (PR still open, but pack manually published to Artifactory)

**Review by gmsantos — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (ilyasemenov84) [reply Dockerfile:30]: Agree, I'll do this before merge.

**Comment by ilyasemenov84** [comment]
@gmsantos 
```
ERROR: Could not find a version that satisfies the requirement biautomation<3.0.0,>=2.0.0 (from versions: 1.1.0)
ERROR: No matching distribution found for biautomation<3.0.0,>=2.0.0
```

**Comment by gmsantos** [comment]
Fixed it @ilyasemenov84

**Review by gmsantos — APPROVED** [review]

**Pull request merged**